internal/txapp/txapp.go

```diff
@@ -205,6 +206,29 @@ func (r *TxApp) ChainInfo(ctx context.Context) (int64, []byte, error) {
 	return height, appHash, nil
 }
 
+// ReloadCache reloads the database state into the engine.
+func (r *TxApp) ReloadCache(ctx context.Context) error {
```

Comment:
I'm personally confused by the name `ReloadCache`.  The fields of TxApp and Engine that are now incorrect and need updating are a lot more than "cache" IMO.  They're just the working state of the application.  I guess let this one alone, but `Reload` seemed fine to me.  "Cache" refers to some low-latency storage of data for performance optimization, where entries can expire or be evicted, and where cache misses result in more costly retrieval from storage etc.  That's not analogous to what we're doing here.  We are refreshing indispensable application state.  e.g. We could achieve a similar effect by creating a new *instance* of Engine via it's constructor where it loads datasets.  Or a more broad reinstantiation of everything, but that's just harder to engineer.